### PR TITLE
feat: allow CLI to find a different port - WF-76

### PR DESCRIPTION
**Pull request by FabienArcellier**
implement #567 

### writer run

image

### writer edit

image

### does not look for the next port when port is set manually

image

### CODE REVIEW

File: src/writer/serve.py [hunk 1]
```diff
@@ -632,3 +630,34 @@ def wf_root_static_assets() -> List[pathlib.Path]:
...
+    """
+    Searches for a free port in a given range on localhost to start the server
+
+    >>> port = _next_localhost_available_port((3005, 3099))
+
+    3005 is the first port to be tested. If it is not available, the port 3006 is tested, and so on.
+    """
+    for port in range(port_range[0], port_range[1]):
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.settimeout(1)  # Définit un timeout d'une seconde
```
Comment (ramedina86) [inline src/writer/serve.py:657]: Please no French comments 😅

File: docs/framework/quickstart-tutorial.mdx [hunk 2]
```diff
@@ -63,12 +63,12 @@ This app will be made up of 2 columns, one with controls for our plot and the se
...
 This way, it will take just `1/3` of the screen. Proportions are calculated relative to each other. Each _Column_, by default, has the value of this factor set to 1. So, when we set the left column to 0.5 and the right to 1, we will get a relation between column sizes of 1:2.
 
-![placing elements](./images/quickstart/placing_elements.png)
+![placing eleme./images/quickstart/placing_elements.png)
 
 Now let's add the rest of the components:
 
 - To the right column, add a _Plotly Graph_.
-- To the left:
+- To thnts](e left:
```
Comment (ramedina86) [inline docs/framework/quickstart-tutorial.mdx:71]: What's this? I'm guessing accidentally typed?

File: docs/framework/quickstart-tutorial.mdx [hunk 3]
```diff
@@ -63,12 +63,12 @@ This app will be made up of 2 columns, one with controls for our plot and the se
 
 This way, it will take just `1/3` of the screen. Proportions are calculated relative to each other. Each _Column_, by default, has the value of this factor set to 1. So, when we set the left column to 0.5 and the right to 1, we will get a relation between column sizes of 1:2.
 
-![placing elements](./images/quickstart/placing_elements.png)
+![placing eleme./images/quickstart/placing_elements.png)
```
Comment (ramedina86) [inline docs/framework/quickstart-tutorial.mdx:66]: What happened here to the link?

File: src/writer/serve.py (hunk 1 above)
Comment (FabienArcellier) [reply src/writer/serve.py:657]: the copy paste from ia ... :(

File: docs/framework/quickstart-tutorial.mdx (hunk 2 above)
Comment (FabienArcellier) [reply docs/framework/quickstart-tutorial.mdx:71]: I suppose it's unfortunate drag and drop from pycharm

**Pull request merged**